src/shim.h

```diff
@@ -15,47 +15,88 @@
 #include <stdio.h>
 
 /* The shim would be handling fixed number of predefined fds.
- * This would be signal fd, stdin fd, proxy socket fd and an I/O
- * fd passed by the runtime
+ * This would be signal fd, stdin fd and a proxy socket connection fd.
  */
-#define MAX_POLL_FDS 4
+#define MAX_POLL_FDS 3
 
 struct cc_shim {
 	char       *container_id;
 	int         proxy_sock_fd;
-	int         proxy_io_fd;
-	uint64_t    io_seq_no;
-	uint64_t    err_seq_no;
-	bool        exiting;
+	char       *token;
+	char       *proxy_address;
+	int         proxy_port;
 };
 
-/*
- * control message format
- * | ctrl id | length  | payload (length-8)      |
- * | . . . . | . . . . | . . . . . . . . . . . . |
- * 0         4         8                         length
+/*  0 1 2 3 4 5 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7
```

Comment:
err, is there a '6' missing in that first block of digits in the comment, or am I missing something? :-)